src/wdmp-c.c

```diff
@@ -47,67 +47,82 @@
 /*                             External Functions                             */
 /*----------------------------------------------------------------------------*/
 
-void wdmp_parse_request(char * payload, req_struct **reqObj)
+void wdmp_parse_generic_request(char * payload, PAYLOAD_TYPE payload_type, void **reqObj)
```

Comment:
I don't see any reason to change the `reqObj` to a `void` vs a `req_struct` as all the fields appear to be the same.  Keeping the `req_struct` type eliminates the need to do casting (which is error prone).